### ISSUE TITLE
vuex and localstorage: arrays are note syncing

### ISSUE THREAD

**Issue by taniki** [gridsome/gridsome#727]
I am using `localStorage` with vuex for persitence and have some issues 

This is not working on `gridsome build` because `$store` is only bootstrapped around a `if(isClient){}`.

```
computed: {
  foo: function(){
    return this.$store.state.foo
  }
}
```

This workaround is working but only if `this.$store.state.foo` is not an array.

```
mounted: function(){
  this.foo = this.$store.state.foo
}
```

I have tried various workarounds but it is either failing on build or generate errors trying to insert stuff early.

## Example

On https://playlists.tkd.now.sh, auth logic is working fine but the array of items is not updated when emitting `logout`. You can pick a look at it here https://github.com/taniki/playlists/blob/master/src/store/index.js

**Comment by realtebo** [comment]
use vue.set to set array element and keep it responsive

it's not a problem with gridsome, but it's a well note working mode of vue.

**Comment by taniki** [comment]
@realtebo Are you speaking about this caveat https://vuejs.org/v2/guide/list.html#Caveats ? If yes, this a different issue. I am replacing the whole array.

Found in https://vuejs.org/2016/02/06/common-gotchas/

> Always modify arrays by using an Array instance method, or replacing it entirely. 

That what I am doing.

**Comment by taniki** [comment]
current working workaround: https://github.com/taniki/playlists/commit/73ebf9a603cc379687a4abfcc12e876ff93597b7 based on this article https://dev.to/viniciuskneves/watch-for-vuex-state-changes-2mgj

TLDR; manually add a watcher with a getter proxy

**Comment by taniki** [comment]
I found some documentation about `<ClientOnly></ClientOnly>` in the "Add External Scripts" section and it is working fine now. The only trick is that `client-only` doesn't prevent `computed(){}` to be included and parsed. It forces to make components for parts that won't be server side. Not a real problem (:
